changes.go

```diff
@@ -212,7 +212,11 @@ func (ch *AddServiceChange) GUIArgs() []interface{} {
 	if options == nil {
 		options = make(map[string]interface{}, 0)
 	}
-	return []interface{}{ch.Params.Charm, ch.Params.Service, options, ch.Params.Constraints}
+	storage := ch.Params.Storage
+	if storage == nil {
+		storage = make(map[string]string, 0)
+	}
+	return []interface{}{ch.Params.Charm, ch.Params.Service, options, ch.Params.Constraints, storage}
```

Comment:
Cool, the GUI will be changed to reflect this, and this does not break the API because the GUI is not using this data yet. So I believe this change is ok.